mobile/src/main/java/com/alexstyl/specialdates/util/Notifier.java

```diff
@@ -89,19 +95,66 @@ public void forDailyReminder(ContactEvents events) {
                 );
 
         String title = NaturalLanguageUtils.joinContacts(context, events.getContacts(), 3);
-        String fullText = TextUtils.join(", ", events.getContacts());
 
         NotificationCompat.Builder builder = new NotificationCompat.Builder(context)
                 .setSmallIcon(R.drawable.ic_stat_contact_event)
                 .setContentTitle(title)
                 .setLargeIcon(largeIcon)
-                .setStyle(new NotificationCompat.BigTextStyle().bigText(fullText))
                 .setVisibility(NotificationCompat.VISIBILITY_PRIVATE)
                 .setAutoCancel(true)
-                .setContentText(fullText)
                 .setContentIntent(intent)
+                .setNumber(events.size())
                 .setColor(context.getResources().getColor(R.color.main_red));
 
+        if (events.size() == 1) {
+            ContactEvent event = events.getEvent(0);
+            String msg = "";
+
+            if (EventType.BIRTHDAY.equals(event.getType())) {
+                int age = event.getContact().getBirthday().getAgeOnYear(event.getYear());
+                msg = context.getString(R.string.turns_age, age);
+
+            } else if (EventType.NAMEDAY.equals(event.getType())) {
+                msg = context.getString(R.string.nameday);
+            }
+
+            NotificationCompat.BigTextStyle bigTextStyle = new NotificationCompat.BigTextStyle().bigText(msg);
+            bigTextStyle.setBigContentTitle(title);
+            builder.setContentText(msg);
+
+            builder.setStyle(bigTextStyle);
+
+        } else if (events.getContacts().size() > 1) {
+            NotificationCompat.InboxStyle inboxStyle = new NotificationCompat.InboxStyle();
+            inboxStyle.setBigContentTitle(title);
+
+            for (int i = 0; i < events.size(); ++i) {
+
+                ContactEvent event = events.getEvent(i);
+                Contact contact = event.getContact();
+                String name = contact.getDisplayName().toString();
+
+                String lineFormatted = name;
+
+                if (EventType.BIRTHDAY.equals(event.getType())) {
```

Comment:
Copy Paste! Oh no!
...
Well, ok then. I'm not yet familiar enough with your code so I'm going to leave such a refactoring up to you. Let me add a big TODO and @deprecated so that it can be found later.

I'm going to create an issue as well so that it is not forgotten